src/main/java/org/hmhb/organization/DefaultOrganizationService.java

```diff
@@ -0,0 +1,118 @@
+package org.hmhb.organization;
+
+import javax.annotation.Nonnull;
+import javax.transaction.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.codahale.metrics.annotation.Timed;
+import org.apache.commons.lang3.StringUtils;
+import org.hmhb.audit.AuditHelper;
+import org.hmhb.exception.organization.CannotDeleteOrganizationWithProgramsException;
+import org.hmhb.exception.organization.OrganizationNameRequiredException;
+import org.hmhb.exception.organization.OrganizationNotFoundException;
+import org.hmhb.program.ProgramDao;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import static java.util.Objects.requireNonNull;
+
+@Service
+public class DefaultOrganizationService implements OrganizationService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultOrganizationService.class);
+
+    private final AuditHelper auditHelper;
+    private final OrganizationDao dao;
+    private final ProgramDao programDao;
+
+    @Autowired
+    public DefaultOrganizationService(
+            @Nonnull AuditHelper auditHelper,
+            @Nonnull ProgramDao programDao,
```

Comment:
I'm avoiding a circular dependency with this.  I don't like it, but I want to validate that an Org doesn't have Programs before trying to delete that Org, and I want to validate that an orgId on a Program is valid before saving a Program.

My choices are between letting a slightly more cryptic error escape, having one service know about another service's dao, writing a special purpose query specifically for this, or cooking up something more complicated.  This works for now.